### DISCUSSION TITLE
How can I modify message

### DISCUSSION THREAD

**Discussion by BWAAEEEK** [pyg-team/pytorch_geometric#6659] [Q&A] [answered]
Hi! I'm trying to implement KGATConv for Knowledge Graph Attention Network (KGAT) using MessagePassig in torch_geometric.
I'm running into some issues when it comes to the Information Aggregation step.
Using the gcn aggregator is not a problem, but I'm finding it challenging to use other aggregators like graph sage or bi-interaction for me 😢.
My question is, can I only aggregate information from the neighboring nodes without making any change or additions to the x_i node representation? 

I thought about just using "None" for the "aggr" argument, but I'm not sure if that's the right solution. 😭

**Comment by BWAAEEEK** [comment]
https://github.com/pyg-team/pytorch_geometric/discussions/3576

It seems there was a similar good question here... Those who have similar question check it out!

**Comment by rusty1s** [comment, marked as answer]
Can you clarify what you mean by "My question is, can I only aggregate information from the neighboring nodes without making any change or additions to the x_i node representation?"? You can use `x_i` in order to create messages, such that both source and destination node features are used for aggregation.

**Reply by BWAAEEEK** [reply]
Hi @rusty1s! Thank you for answering my question.
I thought that the result of propagate is (x_ i + message), if I use "add" to aggr argument.
What I want to do is concatenation x_i and message, not adding. (ex. like this one -> x =  x_i || message)
So to do this, I just wonder that it is possible to get just only message feature corresponding to x_i from the result of propagate!

**Reply by rusty1s** [reply]
Ah, I see. No, `aggr` just defines how messages pointing to the same destination node are reduced. If you want to do `x = x + sum over messages`, this would be
```python
h = self.propagate(edge_index, x=x)
return torch.cat([x, h], dim=-1)

**Reply by BWAAEEEK** [reply]
Oh, so you mean that by doing like that, h is not `x_i + sum over message`???
I just want to get only sum over message of neighbor nodes.

**Reply by rusty1s** [reply]
Yes, this corresponds to `h` in the code snippet above.

**Reply by BWAAEEEK** [reply]
https://pytorch-geometric.readthedocs.io/en/latest/tutorial/create_gnn.html

Then the reason why there isn't `x + self.propagate` in that code is because of add_self_loop ???

**Reply by rusty1s** [reply]
Yes, exactly.

**Reply by BWAAEEEK** [reply]
Thank you so much @rusty1s 😭
